transmission.py

```diff
@@ -0,0 +1,201 @@
+import base64
+import json
+import requests
+import torrent
+
+class InvalidTransmissionQuery(Exception):
+    """An exception thrown when a query is improperly structured when passed to the
+    TransmissionClient
+    """
+    pass
+
+class TransmissionClient(object):
+        def _post_request(self, data):
+            """Prepares the request to the Transmission client.
+
+            Sets the authentication for the client, as well as the data to be requested.
+            Resets the X-Transmission-Session-Id as needed.
+            
+            Args:
+            data (json): A json representation of the data to be transmitted/requested.
+
+            Returns:
+            response (requests.Response): The response from the Transmission Daemon.
+            """
+            url = "http://{}:{}/transmission/rpc".format(self.host, self.port)
+            # Post request
+            response = requests.post(url, data=data, auth=(self.username, self.password), 
+                    headers={ "X-Transmission-Session-Id": self.session_id })
+            # Check if the session id needs to be renewed.
+            if response.status_code == 409:
+                self.session_id = response.headers["X-Transmission-Session-Id"]
+                return requests.post(url, data=data, auth=(self.username, self.password),
+                        headers={ "X-Transmission-Session-Id": self.session_id })
+	    return response
+
+        def __init__(self, host, port=9091, username="transmission", password="transmission"):
+            """Initializes the client to communicate with the Transmission API over the network.
+            
+            Args:
+            username (string): The username used for authentication with the Transmission Daemon.
+            password (string): The password used for authentication with the Transmission Daemon.
+            host (string): The hostname where the Transmission Daemon is located.
+            port (int): The port where the Transmisison Daemon is serving.
+            session_id (string): The X-Transmission-Session-Id currently being used to protect against XSRF attacks.
+            """
+	    self.username = username
+	    self.password = password
+            self.host = host
+            self.port = port
+            self.session_id = ""
+
+        # Action Requests 
+        def start_torrents(self, t_ids=[]):
+            """Starts the torrent files of the associated ids passed in.
+
+            Args:
+            t_ids ([]int): The array of torrent ids to begin torrenting.
```

Comment:
This is python not stupid go... 
list or [] is fine.
if you really need to... [int]